### PR TITLE
Adds documentation for implemented administrative routes

### LINKED ISSUE #42: output from admin routes needs to be documented
https://github.com/Nordstrom/xrpc/pull/38#discussion_r157077534

### PR DESCRIPTION
**Pull request by spenserpothier**
Closes #42

### CODE REVIEW

File: src/main/java/com/nordstrom/xrpc/server/AdminHandlers.java [hunk 1]
```diff
@@ -24,6 +24,159 @@
 import java.util.SortedMap;
 
 public class AdminHandlers {
+  /**
+   * Output metrics reporters in JSON format
```
Comment (andyday) [inline src/main/java/com/nordstrom/xrpc/server/AdminHandlers.java]: might be good to show a portion of these in the readme as well.

File: src/main/java/com/nordstrom/xrpc/server/AdminHandlers.java [hunk 2]
```diff
@@ -73,6 +250,11 @@ public static Handler healthCheckHandler(
             Recipes.ContentType.Application_Json);
   }
 
+  /**
+   * Readiness handler, for use with kubernetes or ELB healthchecks
```
Comment (andyday) [inline src/main/java/com/nordstrom/xrpc/server/AdminHandlers.java]: need to distinguish between this and the ping....

File: README.md [hunk 3]
```diff
@@ -63,6 +63,17 @@ $ java -cp app.jar com.nordstrom.xrpc.demo.DinoGetRequestEncoder trex | curl -k
...
 ```
 
+# Admin routes
+
+xrpc comes with some built in admin routes
+
+* `/metrics` -> Returns the metrics reporters in JSON format
+* `/health` -> Expose a summary of downstream health checks
+* `/ping` -> Responds with a 200-OK status code and the text 'PONG'
+* `/ready` -> should expose a Kubernetes or ELB specific healthcheck for liveliness
```
Comment (andyday) [inline README.md]: Capitalize `S`.  Also need more clarification on the difference between this and health and ping

File: README.md [hunk 4]
```diff
@@ -63,6 +63,17 @@ $ java -cp app.jar com.nordstrom.xrpc.demo.DinoGetRequestEncoder trex | curl -k
...
+* `/restart` -> restart service (should be restricted to approved devs / tooling)
```
Comment (andyday) [inline README.md]: Capitalize `R`

File: README.md [hunk 5]
```diff
@@ -63,6 +63,17 @@ $ java -cp app.jar com.nordstrom.xrpc.demo.DinoGetRequestEncoder trex | curl -k
...
+* `/killkillkill` -> shutdown service (should be restricted to approved devs / tooling)
```
Comment (andyday) [inline README.md]: Capitalize `K`

File: src/main/java/com/nordstrom/xrpc/server/AdminHandlers.java [hunk 6]
```diff
@@ -24,6 +24,159 @@
...
+   *   "meters": {
+   *     "Hard Rate Limits": {
+   *       "count": 0,
+   *       "m15_rate": 0.0,
+   *       "m1_rate": 0.0,
+   *       "m5_rate": 0.0,
+   *       "mean_rate": 0.0,
+   *       "units": "events/second"
+   *     },
+   *     "com.nordstrom.xrpc.server.Router.requests.Rate": {
```
Comment (andyday) [inline src/main/java/com/nordstrom/xrpc/server/AdminHandlers.java]: why do we have namespacing for this and not the others.  this should be requests.Rate

File: src/main/java/com/nordstrom/xrpc/server/AdminHandlers.java [hunk 7]
```diff
@@ -24,6 +24,159 @@
...
+   *   "version": "3.1.3",
+   *   "gauges": null,
+   *   "counters": {
+   *     "com.nordstrom.xrpc.server.Router.Active Connections": {
+   *       "count": 1
+   *     }
+   *   },
+   *   "histograms": null,
+   *   "meters": {
+   *     "Hard Rate Limits": {
```
Comment (andyday) [inline src/main/java/com/nordstrom/xrpc/server/AdminHandlers.java]: this should follow the naming of the other meters below.

File: src/main/java/com/nordstrom/xrpc/server/AdminHandlers.java [hunk 8]
```diff
@@ -24,6 +24,159 @@
...
+   *       "count": 0,
+   *       "m15_rate": 0.0,
+   *       "m1_rate": 0.0,
+   *       "m5_rate": 0.0,
+   *       "mean_rate": 0.0,
+   *       "units": "events/second"
+   *     }
+   *   },
+   *   "timers": {
+   *     "Request Latency": {
```
Comment (andyday) [inline src/main/java/com/nordstrom/xrpc/server/AdminHandlers.java]: again naming mismatch

**Comment by andyday** [comment]
minor fixes.  for the metric naming problems, at least open an issue...

File: README.md [hunk 9]
```diff
@@ -63,6 +63,17 @@ $ java -cp app.jar com.nordstrom.xrpc.demo.DinoGetRequestEncoder trex | curl -k
 trexblue
 ```
 
+# Admin routes
+
+xrpc comes with some built in admin routes
```
Comment (jkinkead) [inline README.md]: You should link to the implementation here, possibly for each route (although line-number links get stale quickly).

File: README.md (hunk 3 above)
Comment (spenserpothier) [reply README.md]: I think the difference between `/health` and `/ping` is explained, but the difference between `/ping` and `/ready` is still unclear. 
As currently implemented I don't think there is a difference, but I would picture it being similar to what is described [here](https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#when-should-you-use-liveness-or-readiness-probes)

Comment (andyday) [reply README.md]: maybe provide a link to this ...

**Comment by spenserpothier** [comment]
@andyday -
Opened issue for naming inconsistencies.
Fixed capitalization misses.
I believe ping/ready differences will be addressed in #60 (and/or #52 )

**Comment by andyday** [comment]
👍 good to go...

**Pull request merged**